daemon/container_operations.go

```diff
@@ -719,6 +719,13 @@ func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName
 		return err
 	}
 
+	if !container.Managed {
```

Comment:
@dongluochen thats not my concern. But I went through the libnetwork code to understand the behavior and I think that is fine. The question I raised wont be impacted here since `addToCluster` will not fail for endpoints connected to networks that `!isClusterEligible()`.